test/framework/src/main/java/org/elasticsearch/common/settings/MockSecureSettings.java

```diff
@@ -36,7 +36,7 @@
  */
 public class MockSecureSettings implements SecureSettings {
 
-    private Map<String, SecureString> secureStrings = new HashMap<>();
+    private Map<String, String> secureStrings = new HashMap<>();
```

Comment:
Just fixing a bug here that prevented node restarts in tests for nodes that used secure settings. If we always return the same `SecureString` for `getString` then if a node ever closes that `SecureString` it can't get the setting again.